arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -436,6 +451,88 @@ def rule_fr_nl_5_06(
+    - No font smaller than "3" (font size can be altered in many ways that depend on browser support)
+    - No white text on white background (many potential "color" values are not easily readable)
+    """
+    invalidTypeFacts = []
+    for fact in val.modelXbrl.facts:
+        validType = fact.concept.instanceOfType(pluginData.formattedExplanationItemTypeQn)
+        wrappedContent = f'<div>{fact.textValue}</div>'  # Single root element required, wrap in valid div element
+        invalidTags = set()
+        invalidStyles = set()
+        hasElts = False
+        for elt in etree.fromstring(wrappedContent).iter():
+            if hasElts and not validType:
+                # `hasElts` skips over wrapping "div" element.
+                #  If additional elements are found and this fact is of an invalid type, trigger an error
+                invalidTypeFacts.append(fact)
+                break
+            hasElts = True
+            tag = elt.tag
+            if tag not in XHTML_ALLOWED_TAGS:
+                # Collect for single error generated later, move on.
+                invalidTags.add(tag)
+                continue
+            if tag in XHTML_ALLOWED_TYPES:
+                typeAttr = elt.get('type')
+                if typeAttr not in XHTML_ALLOWED_TYPES[tag]:
+                    invalidTags.add(f'{tag} (type:"{typeAttr}")')
+                    continue
+            styleAttr = elt.get('style')
+            if styleAttr:
+                styles = [x.split(':') for x in styleAttr.split(';')]
+                for styleValues in styles:
+                    if len(styleValues) > 1:
+                        styleProperty = styleValues[0].strip()
+                        if styleProperty not in XHTML_ALLOWED_STYLES:
+                            invalidStyles.add(styleProperty)
+        # Generate tag/styling errors per-fact so helpful information about fact contents can be reported
+        if len(invalidTags) > 0:
+            yield Validation.error(
+                codes='NL.FR-NL-5.11',
+                msg=_('Only a limited set of XHTML tags (see manual) are allowed in escaped XHTML. '
+                      'Found invalid tags: %(invalidTags)s'),
+                invalidTags=sorted(invalidTags),
+                modelObject=fact
+            )
+        if len(invalidStyles) > 0:
+            yield Validation.error(
+                codes='NL.FR-NL-5.11',
+                msg=_('Only a limited set of style properties (%(allowedStyles)s) are allowed in escaped XHTML. '
+                      'Found invalid style properties: %(invalidStyles)s'),
+                allowedStyles=', '.join(sorted(XHTML_ALLOWED_STYLES)),
+                invalidStyles=sorted(invalidStyles),
+                modelObject=fact
+            )
+    if len(invalidTypeFacts) > 0:
+        yield Validation.error(
+            codes='NL.FR-NL-5.11',
+            msg=_('Formatting using "escaped XHTML" elements MAY ONLY be included in '
```

Comment:
It seems wrong to "speculatively" parse all facts rather than only `formattedExplanationItemType`.  It seems like this should be at most a warning if not removed altogether...

Reply:
I downgraded to a warning. Based on the wording of the rule, I think it's important that the validation check and communicate in some way that certain elements shouldn't have XHTML, but I agree an error is heavy handed based on the speculation.